libr/debug/p/native/linux/linux_debug.c

```diff
@@ -509,3 +514,1397 @@ RList *linux_desc_list (int pid) {
+        h->e_phoff = ph_offset;                 /* Program header table's file offset */
+        h->e_phentsize = ph_size; 
+        h->e_phnum = n_segments + 1;            /* n_segments  + NOTE segment */
+        h->e_flags = 0x0;
+        /* Coredump contains no sections */
+        h->e_shoff = 0x0;
+        h->e_shentsize = 0x0;
+        h->e_shnum = 0x0;
+        h->e_shstrndx = 0x0;
+
+        return h;
+}
+
+static int get_n_mappings(linux_map_entry_t *me_head) {
+        linux_map_entry_t *p;
+        int n_entries;
+
+        for (n_entries = 0, p = me_head; p != NULL ; p = p->n)
+                if ((p->perms & R_MEM) || (p->perms & W_MEM))            /* We don't count maps which does not have r/w perms */
+                        n_entries++;
+
+        return n_entries;
+}
+
+
+static bool dump_elf_header(RBuffer *dest, Elf64_Ehdr *hdr) {
+        bool ret;
+
+	ret = r_buf_append_bytes (dest, (const ut8*)hdr, hdr->e_ehsize);
+	if (ret != true) {
+		perror ("dump_elf_header: error");
+	}
+
+        return ret;
+}
+
+
+static void *get_nt_data(linux_map_entry_t *head, size_t *nt_file_size) {
+        char *maps_data;
+        char *pp;
+        linux_map_entry_t *p;
+        size_t size;            /* global size */
+        unsigned long long n_segments;
+        unsigned long long n_pag;
+
+        size = mapping_file.size;
+        eprintf ("get_nt_size: %ld\n", size);
+        n_segments = mapping_file.count;
+        eprintf ("n_segments: %lld\n", n_segments);
+        n_pag = 1;
+
+        maps_data = malloc (size);
+	if (!maps_data)
+		return NULL;
+
+        pp = maps_data;
+
+        memcpy (maps_data, &n_segments, sizeof (n_segments));
+        memcpy (maps_data + sizeof (n_segments), &n_pag, sizeof (n_pag));
+        pp += sizeof (n_segments) + sizeof (n_pag);
+
+        for (p = head; p != NULL; p = p->n) {
+                if (p->name && strcmp (p->name, "[vdso]") != 0
+                                && strcmp (p->name, "[vsyscall]") != 0
+                                && strcmp (p->name, "[vvar]") != 0
+                                && strcmp (p->name, "[stack]") != 0) {
+                        eprintf ("get_nt_data: %s\n", p->name);
+
+                        memcpy (pp, &p->start_addr, sizeof(p->start_addr));
+                        pp += sizeof (p->start_addr);
+
+                        memcpy (pp, &p->end_addr, sizeof(p->end_addr));
+                        pp += sizeof (p->end_addr);
+
+                        memcpy (pp, &p->offset, sizeof(p->offset));
+                        pp += sizeof (p->offset);
+                }
+        }
+
+        for (p = head; p != NULL; p = p->n) {
+                if (p->name && strcmp (p->name, "[vdso]") != 0
+                                && strcmp (p->name, "[vsyscall]") != 0
+                                && strcmp (p->name, "[vvar]") != 0
+                                && strcmp (p->name, "[stack]") != 0) {
+			memcpy (pp, p->name, p->s_name);
```

Comment:
why not just strncpy, this is just the length of the string. can we kill this var?